runtime/common/src/main/kotlin/kotlinx/serialization/json/Json.kt

```diff
@@ -151,3 +158,34 @@ public class Json(
             plain.parse(deserializer, string)
     }
 }
+
+/**
+ * Builder to conveniently build Json instances.
+ * Properties of this builder are directly matched with properties of [JsonConfiguration].
+ */
+public class JsonBuilder {
+    public var encodeDefaults: Boolean = true
```

Comment:
Using this builder is the same as using an old json constructor with all arguments, just replace `( )` with `{ }`. I'm not sure if this is advantage or drawback, but it seems as overkill to me (discussable)

Reply:
`{}` has multiple advantages compared with the constructor:
 - Easier to read: `Json { foo = true }` vs `Json(JsonConfiguration(foo=true))`
 - Easier to provide migrations for builders while preserving source compatibility and deprecation cycles
 - Builder can be more flexible, e.g. in the face of custom values for class discriminators